src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -468,9 +468,6 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 						 * sample.
 						 */
 						tmp_sample[i] = devc->sample[j++];
-					} else if (devc->capture_flags & CAPTURE_FLAG_DEMUX && (i > 2)) {
-						/* group 2 & 3 get added to 0 & 1 */
-						tmp_sample[i - 2] = devc->sample[j++];
```

Comment:
I wondered if the original author maybe intended to write "|=" instead of "=" because he said "gets added". Still, I do wonder what purpose that would solve. I haven't verified your tests, but I'd also believe that bytes 2+3 are all zero if we halve the channels and then this else-if can really go away.

Reply:
I mean, if a channel group is disabled, I think `tmp_sample` should be zero at that position and that no samples should be folded to a different position, either.